### PR TITLE
fix: Suppressed ssh scheme url warn log

### LINKED ISSUE #9440: Could not parse repo URL
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

SSH URL parsing of GitHub repos is failing.

<br class="Apple-interchange-newline">

**To Reproduce**

1. Install Argo CD 2.4 RC
2. Add a Git repo with SSH URL
3. Deploy applications that reference the added repo
4. Monitor Argo CD repo server logs
5. You should see warnings like following

```
time="2022-05-18T08:06:28Z" level=warning msg="Could not parse repo URL 'git@github.com:my-account/my-repo.git': parse \"git@github.com:my-account/my-repo.git\": first path segment in URL cannot contain colon"
```

**Expected behavior**

No warnings in the logs. Current warnings are confusing and leave a user with an impression as if something is broken.

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

v2.4.0+68a0a83

**Logs**

```
Paste any relevant application logs here.
```

```
time="2022-05-18T08:06:28Z" level=warning msg="Could not parse repo URL 'git@github.com:my-account/my-repo.git': parse \"git@github.com:my-account/my-repo.git\": first path segment in URL cannot contain colon"
```

### PR DESCRIPTION
**Pull request by kenchan0130**
Fixes #9440

The resolution of the above issue, i.e., the degrade, has been resolved; the root resolution of https://github.com/argoproj/argo-cd/issues/9440#issuecomment-1165099222 is outside the scope of this PR.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by kenchan0130** [comment]
The `mage / publish (pull_request)` fail does not seem to be related to this change.

File: pkg/apis/application/v1alpha1/repository_types.go [hunk 1]
```diff
@@ -195,30 +195,40 @@ func (repo *Repository) GetHelmCreds() helm.Creds {
...
-	hostname := ""
+	// For git ssh protocol url without ssh://, url.Parse() will fail to parse.
+	// However, no warn log is output since ssh scheme url is a possible format.
+	if ok, _ := git.IsSSHURL(repoURL); ok {
+		return ""
+	}
 
+	hostname := ""
 	// url.Parse() will happily parse most things thrown at it. When the URL
 	// is either https or oci, we use the parsed hostname to receive the cert,
```
Comment (jsoref) [inline pkg/apis/application/v1alpha1/repository_types.go:206]: Not technically in scope, but if you're in the neighborhood:
```suggestion
	// is either https or oci, we use the parsed hostname to retrieve the cert,
```

**Comment by kenchan0130** [comment]
Could you review this PR?

cc: @crenshaw-dev (because it was commented on in the related issue)

**Comment by crenshaw-dev** [comment]
@kenchan0130 just trying to make sure I understand. The tested "invalid" path is in fact a valid `repo.Repo` value, correct? So won't this just mean that constructing the CA path will silently fail when it is potentially needed? In other words, it might push people to use `insecure` when validation fails.

File: pkg/apis/application/v1alpha1/types_test.go [hunk 2]
```diff
@@ -2619,6 +2619,7 @@ func TestGetCAPath(t *testing.T) {
 		"oci://bar.example.com",
 		"bar.example.com",
 		"ssh://foo.example.com",
+		"git@example.com/organization/reponame.git",
```
Comment (jsoref) [inline pkg/apis/application/v1alpha1/types_test.go:2622]: Fwiw, this is not a proper git url.

It should be:
```
git@example.com:organization/reponame.git
```

Comment (kenchan0130) [reply pkg/apis/application/v1alpha1/types_test.go]: Thank you for your review.

I fixed it by https://github.com/argoproj/argo-cd/pull/9836/commits/ed7195cb10199a52cd34d8b1ccf9504973b15a32.

**Comment by kenchan0130** [comment]
@crenshaw-dev Thank you for your comment. I modified the test to fix the problem.

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks for your patience @kenchan0130!

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.10.

**Pull request merged**